### PR TITLE
gh-119786: create folder in cpython repo for internals documentation

### PR DESCRIPTION
**Pull request by iritkatriel**
This PR just creates the folder and the main page. I would like to add documentation sections in separate PRs so that we verify it's up to date before merging.

* Issue: gh-119786

### CODE REVIEW

File: InternalsDoc/index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+# CPython Internals Documentation
+
+The documentation in this fodler is intended for CPython maintainers.
+It described implementation details of CPython, which should not be
```
Comment (brandtbucher) [inline InternalsDoc/index.md]:
```suggestion
The documentation in this folder is intended for CPython maintainers.
It describes implementation details of CPython, which should not be
```

File: InternalsDoc/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+can change between any two CPython versions and should not be assumed
+to hold for other implementations of the Python language.
+
+The core dev team attempts to keep this documentation up to date. If
+it is not, please report that through the
+[issue tracker](https://github.com/python/cpython/issues).
+
+
+
+
```
Comment (brandtbucher) [inline InternalsDoc/index.md]:
```suggestion
```

**Review by gvanrossum — APPROVED** [review]
LGTM, though may I bikeshed on the name? I will forever type InternalDocs… :-)

File: InternalDocs/index.md
Comment (hugovk) [inline InternalDocs/index.md:1]: If you name this `README.md` it'll be rendered at https://github.com/python/cpython/tree/main/InternalDocs

**Pull request merged**